client/starwhale/api/_impl/dataset/builder.py

```diff
@@ -648,11 +648,12 @@ def update(self, row_item: DataRow) -> None:
         self._raise_run_exception()
         self._queue.put(row_item)
 
-        with self._lock:
-            if self._builder is None:
-                _cls = create_generic_cls(self.__iter__)
-                self._builder = _cls(**self._kw)  # type: ignore
-                self.start()
+        if self._builder is None:
+            with self._lock:
```

Comment:
the `lock` context change is useless. Python use GIL, and the lock here will add almost no overhead.